### PR TITLE
Improve extension points to support Event Publishing 

### PR DESCRIPTION
**Pull request by Shaaali**
### Purpose
$subject.

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

**Review by thivi — APPROVED** [review]

File: modules/access-control/src/access-control-context-provider.tsx [hunk 1]
```diff
@@ -230,7 +230,23 @@ export const AccessControlContext: FunctionComponent<PropsWithChildren<AccessCon
                     featureConfig?.organization_roles,
                     featureConfig?.organization_roles?.scopes?.create,
                     allowedScopes
-                )
+                ),     
+                [ AccessControlConstants.EVENT_CONFIG_READ ] : hasRequiredScopes(
```
Comment (thivi) [inline modules/access-control/src/access-control-context-provider.tsx:234]: Did we add the relevant config to the `deployment.config.json` file? And do we have the relevant j2 templating?

Comment (brionmario) [reply modules/access-control/src/access-control-context-provider.tsx:234]: J2 templating and config mapping can be added to the solution which is implementing this event publishing feature (Using the extension model). We provide support for extended feature declarations.

https://github.com/wso2/identity-apps/commit/aaeb789bfad20739fb52ba4105c3e84b30034a4b

**Review by brionmario — DISMISSED** [review]

**Review by kavinda1995 — APPROVED** [review]

File: modules/access-control/src/access-control-context-provider.tsx (hunk 1 above)
Comment (Shaaali) [reply modules/access-control/src/access-control-context-provider.tsx:234]: @thivi yeah here : 
https://github.com/wso2-enterprise/asgardeo-app-extensions/pull/2189/files#diff-adf8116735540d3bca46ace6bcf7c73841e1b5ca8e5e90cd59b881f77ee2a4c7R535

File: apps/console/src/extensions/configs/endpoints.ts [hunk 2]
```diff
@@ -16,15 +16,17 @@
  * under the License.
  */
 
+/* eslint-disable @typescript-eslint/no-unused-vars */
```
Comment (brionmario) [inline apps/console/src/extensions/configs/endpoints.ts:19]: Why are we supressing a warning here?
```suggestion

```

File: apps/console/src/extensions/configs/endpoints.ts [hunk 3]
```diff
@@ -1,7 +1,7 @@
 /**
- * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ * Copyright (c) 2022, WSO2 LLC. (https://www.wso2.com). All Rights Reserved.
```
Comment (thivi) [inline apps/console/src/extensions/configs/endpoints.ts:2]:
```suggestion
 * Copyright (c) 2021, WSO2 LLC. (https://www.wso2.com). All Rights Reserved.
```

File: apps/console/src/extensions/configs/endpoints.ts [hunk 4]
```diff
@@ -17,14 +17,15 @@
...
 
 import { ExtendedFeatureResourceEndpointsInterface } from "./models";
+import { DeploymentConfigInterface } from "../../features/core";
 
 /**
  * Get the resource endpoints for the extended features.
  *
- * @param {string} serverHost - Server Host.
- * @return {ExtendedFeatureResourceEndpointsInterface}
+ * serverHost - Server Host.
```
Comment (thivi) [inline apps/console/src/extensions/configs/endpoints.ts:25]: Shall we only remove the types? We need to specify the params and return.

File: apps/console/src/extensions/configs/endpoints.ts [hunk 5]
```diff
@@ -17,14 +17,15 @@
...
  */
-export const getExtendedFeatureResourceEndpoints = (serverHost: string): ExtendedFeatureResourceEndpointsInterface => {
+export const getExtendedFeatureResourceEndpoints = (_serverHost: string, 
+    _deploymentConfig: DeploymentConfigInterface): ExtendedFeatureResourceEndpointsInterface => {
     
     return {};
```
Comment (thivi) [inline apps/console/src/extensions/configs/endpoints.ts:30]:
```suggestion
    return;
```

File: apps/console/src/features/server-configurations/configs/ui.ts [hunk 6]
```diff
@@ -1,7 +1,7 @@
 /**
- * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ * Copyright (c) 2022, WSO2 LLC. (https://www.wso2.com). All Rights Reserved.
```
Comment (thivi) [inline apps/console/src/features/server-configurations/configs/ui.ts:2]:
```suggestion
 * Copyright (c) 2020, WSO2 LLC. (https://www.wso2.com). All Rights Reserved.
```

**Review by thivi — DISMISSED** [review]

**Review by thivi — APPROVED** [review]

**Review by brionmario — APPROVED** [review]

**Review by brionmario — APPROVED** [review]

**Review by ThaminduDilshan — APPROVED** [review]

**Pull request merged**